diff --git a/scripts/run-upgrade.sh b/scripts/run-upgrade.sh
index 61768b1..4c8b636 100755
--- a/scripts/run-upgrade.sh
+++ b/scripts/run-upgrade.sh
@@ -171,6 +171,7 @@ function main {
         # rebuild the repo servers
         RUN_TASKS+=("repo-install.yml")
         # explicitly perform mariadb upgrade
+        RUN_TASKS+=("${UPGRADE_PLAYBOOKS}/mariadb-shutdown.xml")
         RUN_TASKS+=("galera-install.yml -e 'galera_upgrade=true'")
         # explicitly perform controlled galera cluster restart
         RUN_TASKS+=("${UPGRADE_PLAYBOOKS}/galera-cluster-rolling-restart.yml")
