diff --git a/scripts/run-upgrade.sh b/scripts/run-upgrade.sh
index 7eff51393..33a79180d 100755
--- a/scripts/run-upgrade.sh
+++ b/scripts/run-upgrade.sh
@@ -175,6 +175,7 @@ function main {
         RUN_TASKS+=("setup-hosts.yml --limit '!galera_all'")
         # add new container config to galera containers but don't restart
         RUN_TASKS+=("lxc-containers-create.yml -e 'lxc_container_allow_restarts=false' --limit galera_all")
+        RUN_TASKS+=("${UPGRADE_PLAYBOOKS}/lxc-containers-restart.yml --limit '!galera_all'")
         RUN_TASKS+=("haproxy-install.yml")
 	# rebuild the repo servers
         RUN_TASKS+=("repo-install.yml")
